library/src/hcc_detail/hipsparse.cpp

```diff
@@ -5490,6 +5500,488 @@ hipsparseStatus_t hipsparseZcsrilu02(hipsparseHandle_t         handle,
+
+    return rocSPARSEStatusToHIPStatus(status);
+}
+
+hipsparseStatus_t hipsparseZbsric02_bufferSize(hipsparseHandle_t         handle,
+                                               hipsparseDirection_t      dirA,
+                                               int                       mb,
+                                               int                       nnzb,
+                                               const hipsparseMatDescr_t descrA,
+                                               hipDoubleComplex*         bsrValA,
+                                               const int*                bsrRowPtrA,
+                                               const int*                bsrColIndA,
+                                               int                       blockDim,
+                                               bsric02Info_t             info,
+                                               int*                      pBufferSizeInBytes)
+{
+    if(pBufferSizeInBytes == nullptr)
+    {
+        return HIPSPARSE_STATUS_INVALID_VALUE;
+    }
+
+    size_t           buffer_size;
+    rocsparse_status status;
+
+    status = rocsparse_zbsric0_buffer_size((rocsparse_handle)handle,
+                                           hipDirectionToHCCDirection(dirA), 
+                                           mb,
+                                           nnzb,
+                                           (rocsparse_mat_descr)descrA,
+                                           (rocsparse_double_complex*)bsrValA,
+                                           bsrRowPtrA,
+                                           bsrColIndA,
+                                           blockDim,
+                                           (rocsparse_mat_info)info,
+                                           &buffer_size);
+
+    *pBufferSizeInBytes = (int)buffer_size;
+
+    return rocSPARSEStatusToHIPStatus(status);
+}
+
+hipsparseStatus_t hipsparseSbsric02_analysis(hipsparseHandle_t         handle,
+                                             hipsparseDirection_t      dirA,
+                                             int                       mb,
+                                             int                       nnzb,
+                                             const hipsparseMatDescr_t descrA,
+                                             const float*              bsrValA,
+                                             const int*                bsrRowPtrA,
+                                             const int*                bsrColIndA,
+                                             int                       blockDim,
+                                             bsric02Info_t             info,
+                                             hipsparseSolvePolicy_t    policy,
+                                             void*                     pBuffer)
+{
+    // Obtain stream, to explicitly sync (cusparse bsric02_analysis is blocking)
```

Comment:
Did you verify cusparse is blocking?

Reply:
Their documentation says it does not support asynchronous execution but I did not verify it. I am actually not sure how I would verify it